### ISSUE TITLE
Race condition encountered: attempted to load session[SESSION_ID] which has been created but not yet serialized.

### ISSUE THREAD

**Issue by asundaram** [jcoleman/tomcat-redis-session-manager#72]
We are getting this error in Production while using Tomcat 6 version of the redis session manager. Since **saveOnChange** feature doesn't exist in this branch, we cannot use it (unfortunate). We would like to know what impact this exception has to the users.

In New Relic we see transactions with these errors have a 200 HTTP response code. Does this mean that the user did not receive errors(maybe because of a retry somewhere)? Also there are 100's of such errors everyday in our logs but haven't heard any issues reported from our users or testers.

Please advice.

**Comment by jaychang9** [comment]
I also have this problem,how to solve it?

**Comment by haroldyong** [comment]
yes. i met the same problem-when store large object into session. so i change google MSM . it fine.

**Comment by jaychang9** [comment]
What's google MSM?
Can you give the website address about google MSM.



jaychang@aliyun.com
 
From: 3Y03
Date: 2016-01-05 12:37
To: jcoleman/tomcat-redis-session-manager
CC: Jay Chang
Subject: Re: [tomcat-redis-session-manager] Race condition encountered: attempted to load session[SESSION_ID] which has been created but not yet serialized. (#72)
yes. i met the same problem-when store large object into session. so i change google MSM . it fine.

**Comment by maheshchotrani** [comment]
> yes. i met the same problem-when store large object into session. so i change google MSM . it fine.

@haroldyong , can you please provide with an example, how did solve this. 

Also appreciate if you can elaborate on MSM JAR(s) - `kryo-serializer`, `javolution-serializer`, `xstream-serializer` or any other.

Thanks
